core/src/main/scala/org/bykn/bosatsu/TypeRef.scala

```diff
@@ -86,6 +87,26 @@ object TypeRef {
 
   case class TypeLambda(params: NonEmptyList[TypeVar], in: TypeRef) extends TypeRef
 
+  def fromType(tpe: Type): Option[TypeRef] = {
```

Comment:
this can maybe better be:
```scala
def fromTypeA[F[_]: Apply](tpe: Type, orElse: F[TypeRef])(fn: Type.Const.Defined => F[TypeRef]): F[TypeRef]
```
then we can use it in more contexts.